### PR TITLE
(fix): don't handle time-dtypes as extension arrays in `from_dataframe`

### LINKED ISSUE #9026: Potential regression in Dataset.from_dataframe() not preserving timezone
### What happened?

Converting pandas DataFrame that has a datetime column with timezone to an xarray dataset does not preserve the timezone, this only breaks in version 2024.5

### What did you expect to happen?

I would expect the timezone info to be preserved, as it was the case before.

### Minimal Complete Verifiable Example

```Python
import pandas as pd
import xarray as xr

df1 = pd.DataFrame(
    {"A": pd.date_range("20130101", periods=4, tz="US/Eastern"), "B": [1, 2, 3, 4]}
)
dataset = xr.Dataset.from_dataframe(df1)
df2 = dataset.to_dataframe()

print(df1.dtypes, dataset.dtypes, df2.dtypes, sep="\n\n")
```

### MVCE confirmation

- [X] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [X] Complete example — the example is self-contained, including all data and the text of any traceback.
- [X] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [X] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [X] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Relevant log output

```Python
# On xarary 2024.5.0:
A    datetime64[ns, US/Eastern]
B                         int64
dtype: object

Frozen({'A': dtype('<M8[ns]'), 'B': dtype('int64')})

A    datetime64[ns]
B             int64
dtype: object

# ---------------------------
#  On previous versions:

A    datetime64[ns, US/Eastern]
B                         int64
dtype: object

Frozen({'A': dtype('O'), 'B': dtype('int64')})

A    datetime64[ns, US/Eastern]
B                         int64
dtype: object
```

### Anything else we need to know?

_No response_

### Environment

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.12.1 (tags/v3.12.1:2305ca5, Dec  7 2023, 22:03:25) [MSC v.1937 64 bit (AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
byteorder: little
LC_ALL: None
LANG: None
LOCALE: ('English_United States', '1252')
libhdf5: 1.14.2
libnetcdf: None

xarray: 2024.5.0
pandas: 2.2.2
numpy: 1.26.4
scipy: 1.12.0
netCDF4: None
pydap: None
h5netcdf: None
h5py: 3.10.0
zarr: None
cftime: None
nc_time_axis: None
iris: None
bottleneck: None
dask: None
distributed: None
matplotlib: 3.8.3
cartopy: None
seaborn: None
numbagg: None
fsspec: 2024.3.1
cupy: None
pint: None
sparse: None
flox: None
numpy_groupies: None
setuptools: 69.2.0
pip: 24.0
conda: None
pytest: None
mypy: None
IPython: 8.22.2
sphinx: 7.2.6

</details>

### PR DESCRIPTION
**Pull request by ilan-gold**
This is probably the simplest way to fix the bug but I now have many questions about `PandasIndexingAdapter`.  I initially attempted to make datetime dtypes go into `ExtensionArray` (because they are extension arrays) but there are so many tests with behavior that I don't understand that this seemed easier.  Primarily, my issue with doing that wasn't that the tests broke in a way that was "this behavior doesn't make sense" but rather "this behavior makes _more_ sense than what the test expects" i.e., there were a number of places where tests expect datetime dtypes to _not_ be preserved, but they are preserved by using `ExtensionArray`.  But I don't understand why certain things are expected or not.  For example:

https://github.com/pydata/xarray/blob/026aa7c073d03fb6d749c216be1d829816583fac/xarray/tests/test_variable.py#L2967-L2968

became a condition that expected the `dtype` be preserved and not just blanket converting everything to `np.dtype("datetime64[ns]")` i.e., the `dtype` on `var` remained `pd.DatetimeTZDtype("s", pytz.timezone("America/New_York"))`

- [x] Closes #9026
- [x] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: properties/test_pandas_roundtrip.py [hunk 1]
```diff
@@ -98,3 +108,15 @@ def test_roundtrip_pandas_dataframe(df) -> None:
...
+
+
+@given(df=dataframe_strategy)
+def test_roundtrip_pandas_dataframe_datetime(df) -> None:
+    # Need to name the indexes, otherwise Xarray names them 'dim_0', 'dim_1'.
+    df.index.name = "rows"
+    df.columns.name = "cols"
+    dataset = xr.Dataset.from_dataframe(df)
+    roundtripped = dataset.to_dataframe()
+    roundtripped.columns.name = "cols"  # why?
```
[the remark this answers is not in the document]
Comment (ilan-gold) [reply properties/test_pandas_roundtrip.py:120]: thanks for linking!

**Comment by ilan-gold** [comment]
@dcherian could you review?

File: properties/test_pandas_roundtrip.py [hunk 2]
```diff
@@ -98,3 +108,15 @@ def test_roundtrip_pandas_dataframe(df) -> None:
...
+    pd.testing.assert_frame_equal(df, roundtripped)
```
Comment (dcherian) [inline properties/test_pandas_roundtrip.py:121]: How is this equal if its dropping the timezone info?

Comment (dcherian) [reply properties/test_pandas_roundtrip.py:121]: ok I see now.

**Pull request merged**